[Dingwei] iP

SherrinfordHR opened the pull request:
### Watson

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Watson frees your mind of having to remember things you need to do. It's,

*   text-based
*   easy to learn
*   FAST _SUPER_ FAST to use

All you need to do is,

1.  download it from [here]().
2.  double-click it.
3.  add your tasks.
4.  let it manage your tasks for you 😉

And it is FREE!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

* * *

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```ruby
    public class Main {
        public static void main(String[] args) {
            Application.launch(MainApp.class, args);
        }
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

SherrinfordHR committed:
level 1-0 Rename to Watson, Greet, Exit

SherrinfordHR committed:
level 1-1 Use scanner to read input and while loop till "bye" is read

SherrinfordHR committed:
level 1-2 use ArrayList to store the tasks and use for loop to display them one by one

SherrinfordHR committed:
level 1-3 mark as done by splitting the command if it contains mark and must be size of 2, but there could be loop holes

SherrinfordHR committed:
level 1-3 some bug fixed. Bookmark 2 will not work as mark 2

SherrinfordHR committed:
level 1-4 add different subclass of task

SherrinfordHR committed:
Level-5 but not including 4

SherrinfordHR committed:
level 6 including 5

SherrinfordHR committed:
Level-7 manage to save to the tasklist.txt and read correctly

SherrinfordHR committed:
Level-8 added in time

SherrinfordHR committed:
A-OOP and A-Package

SherrinfordHR committed:
Merge branch 'add-gradle-support' of https://github.com/SherrinfordHR/ip

* 'add-gradle-support' of https://github.com/SherrinfordHR/ip:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

SherrinfordHR committed:
A-Gradle, A-JUnit and A-Jar

SherrinfordHR committed:
A-JavaDoc

SherrinfordHR committed:
correct Java doc format

SherrinfordHR committed:
A-CodingStandard

SherrinfordHR committed:
Level-9

SherrinfordHR committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  A-CodingStandard

SherrinfordHR committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  correct Java doc format
  A-JavaDoc

# Conflicts:
#	src/main/java/Watson/command/Command.java
#	src/main/java/Watson/command/MarkCommand.java
#	src/main/java/Watson/exception/WatsonException.java
#	src/main/java/Watson/storage/Storage.java
#	src/main/java/Watson/task/Deadline.java
#	src/main/java/Watson/task/Events.java
#	src/main/java/Watson/task/Task.java
#	src/main/java/Watson/task/ToDo.java

SherrinfordHR committed:
Minor adjustment

SherrinfordHR committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Level-9

# Conflicts:
#	src/main/java/Watson/parser/Parser.java
#	src/main/java/Watson/task/Task.java

SherrinfordHR committed:
updates on Level-9

wj200 commented on src/main/java/Watson/command/SearchCommand.java:
```diff
@@ -0,0 +1,21 @@
+package Watson.command;
```
Header comment for package is missing

wj200 commented on src/main/java/Watson/ui/Ui.java:
```diff
@@ -113,4 +113,15 @@ public void showMessage(String message) {
     }
+
+    public void showTasks(List<Task> matchingTasks) {
+        if (matchingTasks.isEmpty()){
```
Good adherence to minimising the layer of nesting for the happy path

wj200 reviewed (commented):
Overall LGTM

Perhaps you could include comments and javadocs in the next commit.

WilkinsAng commented on src/main/java/Watson/storage/Storage.java:
```diff
@@ -0,0 +1,104 @@
+    /**
+     * Parses a line from the storage file into a Task object.
+     *
+     * @param line The line read from the file, formatted as "TYPE | STATUS | DESCRIPTION | [EXTRA FIELDS]".
```
I think this is a nice and clear documentation for this function.
Tells me what to expect without having to find the calling function

WilkinsAng commented on src/main/java/Watson/task/Events.java:
```diff
@@ -0,0 +1,77 @@
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+            this.fromdate = LocalDateTime.parse(from, formatter);
+        } catch (DateTimeException e) {
```
Correct me if I am wrong.
Are you still allowing the creation of events even if there is no from and to dates?

WilkinsAng commented on src/main/java/Watson/task/TaskList.java:
```diff
@@ -0,0 +1,85 @@
+     *
+     * @param task The task to add.
+     */
+    public void add(Task task) {
```
Perhaps the naming of these functions can be clearer?
I think names like addTask and getTask will be clearer and more consistent with your later functions like findTasks

WilkinsAng commented on src/main/java/Watson/task/Task.java:
```diff
@@ -0,0 +1,85 @@
+     *
+     * @return The task description (e.g., "Read book").
+     */
+    public String tofile() {
```
Perhaps you can make these functions to camelCase to be in line with the coding standards?

WilkinsAng reviewed (commented):
Overall, LGTM!
I just noticed some of your functions are not in camelCase sprinkled around your code.
Nice and clear javadocs as well.
Jiayous for the rest of 2103!

zlllllr commented on src/main/java/Watson/storage/Storage.java:
```diff
@@ -0,0 +1,104 @@
+package Watson.storage;
+
+import Watson.task.*;
```
Perhaps you can list out the imported classes explicitly to make it clearer.

zlllllr commented on src/main/java/Watson/task/Task.java:
```diff
@@ -0,0 +1,85 @@
+     *
+     * @return The task description (e.g., "Read book").
+     */
+    public String getdescription() {
```
Perhaps you could change the name of the method to camelCase?

zlllllr commented on src/main/java/Watson/parser/Parser.java:
```diff
@@ -0,0 +1,42 @@
+        String[] parts = command.split(" ", 2);
+        String action = parts[0].toLowerCase();
+
+        switch (action) {
```
I like the use of the switch-case statements here, it makes the code very easy to understand!

zlllllr reviewed (commented):
LGTM! Overall, your code is pretty clean and logical. There is only some small naming and importing issues and everything else looks good! All the best to your 2103!

SherrinfordHR committed:
Level-10 making GUI with minimal change in my code

SherrinfordHR committed:
The codebase lacks runtime checks for critical internal assumptions, allowing potential invalid states to propagate silently.

Undetected invalid states (e.g., null/empty keywords, malformed inputs) may lead to undefined behavior or subtle bugs that are difficult to trace during later development stages.

Add assertions to enforce invariants at key interaction points:
- Validate non-null/non-empty keywords in SearchCommand
- Verify input format correctness in AddCommand
- Guard against invalid indices in MarkCommand
- Check status flag validity during task deserialization
- Ensure non-null task lists in UI methods

Assertions provide fail-fast behavior during testing/development while avoiding production overhead. They explicitly document assumptions and surface errors closer to their origin, simplifying debugging.

SherrinfordHR committed:
Inconsistent naming conventions, missing input validations and poor strucutre

Non-standard method names, redundant code and arrow head structure could hinder readability and scalability

Fix Event task typo from [D] to [E]
Rename method names
Validate command arguments

Reduce runtime exception and adhere to Java naming convention to enhance code maintainability

SherrinfordHR committed:
Merge branch 'branch-A-Assertions'

* branch-A-Assertions:
  The codebase lacks runtime checks for critical internal assumptions, allowing potential invalid states to propagate silently.

# Conflicts:
#	src/main/java/Watson/command/AddCommand.java
#	src/main/java/Watson/task/Task.java

SherrinfordHR committed:
Tasks lack priority, limiting user's ability to categorize them

User needs to focus and schedule the tasks according to their urgency

Introduce a Priority enum with HIGH, MEDIUM, LOW.
Add priority field to Task class and propagate to subclasses (ToDo, Deadline, Event).
Update AddCommand and Parser to parse /priority flag from user input.
Modify Storage to serialize/deserialize priority in task data.
Extend toString() and UI methods to display task priority.
Add PriorityCommand to update priorities of existing tasks.

Backward compatibility is maintained by defaulting to MEDIUM if priority is missing in stored tasks.
Explicit /priority flag in commands prevents ambiguity with task descriptions.

SherrinfordHR committed:
Minor update to show welcome message when booted

SherrinfordHR committed:
Ui.png added in docs file

SherrinfordHR committed:
Change the current README.md

SherrinfordHR committed:
Change both user and bot image

SherrinfordHR committed:
Update Ui.png

SherrinfordHR committed:
Change the filepath so it it relate filepath not absolute
